src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -258,7 +249,67 @@ public ITelemetryChannel TelemetryChannel
         /// <remarks>
         /// This feature is opt-in and must be configured to be enabled.
         /// </remarks>
-        public IApplicationIdProvider ApplicationIdProvider { get; set; }
+        public IApplicationIdProvider ApplicationIdProvider
+        {
+            get
+            {
+                return this.applicationIdProvider;
+            }
+
+            set
+            {
+                this.applicationIdProvider = value;
+                SetApplicationIdEndpoint(this.applicationIdProvider, this.Endpoint.FormattedApplicationIdEndpoint);
+            }
+        }
+
+        /// <summary>
+        /// Gets the Endpoint Controller responsible for making service endpoints available.
+        /// </summary>
+        public EndpointContainer Endpoint { get; private set; } = new EndpointContainer(new EndpointProvider());
```

Comment:
rename variable to say EndPointContainer

Reply:
Can you explain why?
I don't have a strong opinion, but I felt that "Endpoint" would keep it simpler

Reply:
done

Reply:
so that variable name hints about its type.